src/cgame/cg_players.cpp

```diff
@@ -2733,7 +2733,7 @@ void CG_Player( centity_t *cent )
 	int           renderfx;
 	entityState_t *es = &cent->currentState;
 	class_t       class_ = (class_t) ( ( es->misc >> 8 ) & 0xFF );
-	float         scale;
+	const classModelConfig_t *cmc = BG_ClassModelConfig( class_ );
```

Comment:
You should use ci->modelScale

Reply:
The `ci->modelScale` value comes from the `character.cfg`, not from the `.model.cfg` file.

The `ci->modelScale` value is then sent to the engine for the in-engine scaling.

Setting a non-1 scale in both `character.cfg` and `cmc->modelScale` will probably multiply the scales, we may do the multiplication in cgame and delete the scaling code in engine, but that is out of scope of this PR.

Reply:
The model code is messed up right now. I recommend sticking with the status quo for now instead of having multiplied scales. We can refactor and bring consistency later.

Reply:
I'm not saying we want to multiply the scale, I'm just guessing what is happning if both mechanism exist.

Actually the modelScale in `character.cfg` looks like a hack. Quoting the human model `character.cfg`:

```
// modelScale is used to customize player models made originally for Quake4
// Quake4 player models are taller than Q3A player models and need to be sized down
// by a factor of 0.83 in each coordinate axis
// if modelScale is not specified it defaults to 1 1 1
```

There is no `modelScale` in ioq3 neither in etlegacy. It looks to be a quirk inherited from XreaL. Doing it in `character.cfg` is hacky, we better do it in `.model.cfg` files like buildables and missiles already do. The implementation itself is hacky.

Actually once this is merged I will probably go next by NUKING the `character.cfg`'s  modelScale feature from both cgame and engine (and then multiplication will not be a concern).

Reply:
Then let's bring zOffset too, since that is a model characteristic in the character.cfg